### PR TITLE
add support for loading a customAuthenticator class #1001 - needs cli…

### PR DESCRIPTION
**Pull request by gtully**
…ent_java 1.4

requires https://github.com/prometheus/client_java/issues/1088

### CODE REVIEW

**Comment by gtully** [comment]
Is there a branch where the snapshot versions are referenced? I have tweaked the pom.xml (which cannot correct) to have this work. thanks!

**Comment by dhoard** [comment]
The main branch build version is generic `1.x.x` (no `-SNAPSHOT` version.)

File: pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <prometheus.metrics.version>1.3.1</prometheus.metrics.version>
+        <prometheus.metrics.version>1.4.0-SNAPSHOT</prometheus.metrics.version>
```
Comment (gtully) [inline pom.xml:27]: @dhoard thanks, for this change, I guess there is no other option, the client_java project uses snapshots.

Comment (dhoard) [reply pom.xml:27]: @gtully to clarify, by design, the `jmx_exporter` project only uses `client_java` release artifacts. There is no snapshot dependency between the projects.

It does become cumbersome when there are dependencies, but it only occurs rarely.

Comment (gtully) [reply pom.xml:27]: thanks, that makes sense, this will have to wait for https://github.com/prometheus/client_java/issues/1088

**Comment by dhoard** [comment]
@gtully Thanks for the PR!!! I merged the code/made an update to fix the build.

**Comment by gtully** [comment]
@dhoard thanks for taking the time to review and merge, much appreciated.

**Pull request merged**